shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java

```diff
@@ -135,7 +135,15 @@ public ConsoleSessionImpl(SessionFactory factory,
         }
 
         // Create session
-        session = processor.createSession(in, out, err);
+        if (in == null) {
```

Comment:
Just in case, it's maybe better to test `out` and `err` as well:

```
if (in == null || out == null || err == null) {
```

Reply:
Yes, sorry for the mistake, i have push a new commit